1/main.py

```diff
@@ -0,0 +1,133 @@
+def shape(m):
+    assert m, "map must be not empty"
+    return len(m), len(m[0])
+
+
+def printer(ch, curr, max_index):
+    if curr == max_index - 1:
+        print(ch)
+    else:
+        print(ch, end='')
+
+
+def print_map(m, pos):
+    height = len(m)
+    length = len(m[0])
+    assert pos[0] <= height - 1 and pos[1] <= length - 1, "Indiana Jones left the world"
+    for row in range(height):
+        for col in range(length):
+            if (row, col) == pos:
+                printer('@', col, length)
+            elif m[row][col] is False:
+                printer('#', col, length)
+            else:
+                printer('.', col, length)
+
+
+def neighbours(m, pos):
+    result = []
+    height = len(m)
+    length = len(m[0])
+    assert pos[0] < height and pos[1] < length, "off-map position"
+    if pos[0] < height - 1 and m[pos[0] + 1][pos[1]]:
+        result.append((pos[0] + 1, pos[1]))
+    if pos[0] > 0 and m[pos[0] - 1][pos[1]]:
+        result.append((pos[0] - 1, pos[1]))
+    if pos[1] < length - 1 and m[pos[0]][pos[1] + 1]:
+        result.append((pos[0], pos[1] + 1))
+    if pos[1] > 0 and m[pos[0]][pos[1] - 1]:
+        result.append((pos[0], pos[1] - 1))
+    # print(result)
```

Comment:
You can leave comments that explain the code or some reminders like `# TODO: change to ...`, but if the code is ready, it's better to remove such debug comments. You can use the `logging` library and log everything you want as `DEBUG` level or `INFO`, and switch between them when launching the script, without removing "logging code" (instead of "printing code") from your functions.